bindings/go/plugin/manager/registries/componentversionrepository/converter.go

```diff
@@ -0,0 +1,230 @@
+	request := ocmrepositoryv1.GetLocalResourceRequest[runtime.Typed]{
+		Repository: c.repositorySpecification,
+		Name:       component,
+		Version:    version,
+		Identity:   identity,
+	}
+
+	response, err := c.externalPlugin.GetLocalResource(ctx, request, c.credentials)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	rBlob, err := c.createBlobData(response.Location)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create blob data: %w", err)
+	}
+
+	convert := descriptor.ConvertFromV2Resources([]descriptorv2.Resource{*response.Resource})
+
+	return rBlob, &convert[0], nil
+}
+
+func (c *componentVersionRepositoryWrapper) AddLocalSource(ctx context.Context, component, version string, res *descriptor.Source, content blob.ReadOnlyBlob) (*descriptor.Source, error) {
+	sources, err := descriptor.ConvertToV2Sources(c.scheme, []descriptor.Source{*res})
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert source: %w", err)
+	}
+
+	tmp, err := os.CreateTemp("", "source")
+	if err != nil {
+		return nil, fmt.Errorf("failed to create temp file: %w", err)
+	}
+	reader, err := content.ReadCloser()
+	if err != nil {
+		return nil, fmt.Errorf("failed to read content: %w", err)
+	}
+
+	defer func() {
+		_ = tmp.Close()
+		_ = reader.Close()
+	}()
+
+	if _, err := io.Copy(tmp, reader); err != nil {
+		return nil, fmt.Errorf("failed to copy content: %w", err)
+	}
+
+	request := ocmrepositoryv1.PostLocalSourceRequest[runtime.Typed]{
+		Repository: c.repositorySpecification,
+		Name:       component,
+		Version:    version,
+		Source:     &sources[0],
+		SourceLocation: types.Location{
+			LocationType: types.LocationTypeLocalFile,
+			Value:        tmp.Name(),
+		},
+	}
+
+	return c.externalPlugin.AddLocalSource(ctx, request, c.credentials)
+}
+
+func (c *componentVersionRepositoryWrapper) GetLocalSource(ctx context.Context, component, version string, identity runtime.Identity) (blob.ReadOnlyBlob, *descriptor.Source, error) {
+	request := ocmrepositoryv1.GetLocalSourceRequest[runtime.Typed]{
+		Repository: c.repositorySpecification,
+		Name:       component,
+		Version:    version,
+		Identity:   identity,
+	}
+
+	response, err := c.externalPlugin.GetLocalSource(ctx, request, c.credentials)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	rBlob, err := c.createBlobData(response.Location)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create blob data: %w", err)
+	}
+
+	convert := descriptor.ConvertFromV2Sources([]descriptorv2.Source{*response.Source})
+
+	return rBlob, &convert[0], nil
+}
+
+func (c *componentVersionRepositoryWrapper) createBlobData(location types.Location) (blob.ReadOnlyBlob, error) {
+	var rBlob blob.ReadOnlyBlob
+
+	if location.LocationType == types.LocationTypeLocalFile {
+		file, err := os.Open(location.Value)
+		if err != nil {
+			return nil, err
+		}
+
+		rBlob = blob.NewDirectReadOnlyBlob(file)
+	}
```

Comment:
I think this is missing error cases

Reply:
also you should use a file blob for this, not a read only blob, because you dont want to buffer the file content permanently in memory. (i know the name of the function is misleading, in blob 0.0.3 it was deprecated because of that)

Reply:
>(i know the name of the function is misleading, in blob 0.0.3 it was deprecated because of that)

Oh I see. I thought it's gonna be a wrapper.